testsuite/features/support/file_management.rb

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2019-2023 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+# Attempts to retrieve the SHA256 checksum for a file inside a given directory or download it
+# from the same domain the file has been downloaded from.
+# Returns the path to the checksum or nil if none has been found/downloaded
+def get_checksum_path(dir, original_file_name, file_url)
+  checksum_file_names = %W[CHECKSUM SHA256SUMS sha256sum.txt #{original_file_name}.CHECKSUM #{original_file_name}.sha256]
+
+  # When using a mirror, the checksum file should be present and in the same directory of the file
+  if $mirror
+    checksum_file_names.each do |name|
+      checksum_path = "#{dir}/#{name}"
+      _output, code = get_target('server').run("test -e #{checksum_path}", check_errors: false)
+      return checksum_path if code.zero?
+    end
+  # Attempt to download the checksum file
+  else
+    base_url = file_url.delete_suffix(original_file_name)
+
+    checksum_file_names.each do |name|
+      checksum_url = base_url + name
+      _output, code = get_target('server').run("cd #{dir} && wget --no-check-certificate #{checksum_url}", check_errors: false, timeout: 1500)
```

Comment:
since you have an if else, and both need the server, I would just create a variable at the beginning of the function to get the server as a target and then use that :thinking:

Reply:
Good catches, thanks !